In [1]:
import pandas as pd
import numpy as np

In [2]:
pd_data = pd.read_csv("data/iris.csv")
pd_data.shape

(12, 2)

In [3]:
pd_data

Length  Class
0      1.4      0
1      1.0      0
2      1.3      0
3      1.9      0
4      2.0      0
5      1.8      0
6      3.0      1
7      3.8      1
8      4.1      1
9      3.9      1
10     4.2      1
11     3.4      1

**Câu hỏi 11: Giá trị mean và variance của biến đầu vào (Length) cho "Class"="0" lần lượt là:**

In [4]:
vect_length = pd_data.query("Class == 0")["Length"]
vect_length.mean()

np.float64(1.5666666666666667)

In [5]:
# variance
np.var(vect_length)

np.float64(0.1288888888888889)

In [6]:
# 11 -> a)

**Câu hỏi 12: Giá trị mean và variance của biến đầu vào (Length) cho "Class"="1" lần lượt là:**

In [7]:
vect_length = pd_data.query("Class == 1")["Length"]
vect_length.mean()

np.float64(3.733333333333333)

In [8]:
# variance
np.var(vect_length)

np.float64(0.1722222222222222)

In [9]:
# 12 -> b)

**Câu hỏi 13: Cho dữ liệu kiểm thử X = (Length=3.4). Xác suất dữ liệu kiểm thử X thuộc vào "Class"="0", "Class"="1"?**